### PR TITLE
update env variable for app insights and adjust tests accordingly

### PR DESCRIPTION
**Pull request by arroyc**
ready

### CODE REVIEW

File: tests/Oryx.RuntimeImage.Tests/Node/NodeRuntimeImageCanRunWhenAppInsightsModuleNotFound.cs [hunk 1]
```diff
@@ -86,11 +86,11 @@ public async Task GeneratesScript_CanRun_AppInsights_NotConfigured(string nodeVe
             var volume = DockerVolume.CreateMirror(hostDir);
             var appDir = volume.ContainerDir;
             var imageName = string.Concat("oryxdevmcr.azurecr.io/public/oryx/node-", nodeVersion);
-            var aIEnabled = "APPLICATIONINSIGHTSAGENT_EXTENSION_ENABLED";
+            var aIEnabled = "ApplicationInsightsAgent_EXTENSION_VERSION";
             int containerDebugPort = 8080;
 
             var script = new ShellScriptBuilder()
-                .AddCommand($"export {aIEnabled}=TRUE")
+                .AddCommand($"export {aIEnabled}=disabled")
```
Comment (markwolff) [inline tests/Oryx.RuntimeImage.Tests/Node/NodeRuntimeImageCanRunWhenAppInsightsModuleNotFound.cs]: I think this test will fail. You can make this `~2` to simulate the actual value.

File: src/startupscriptgenerator/src/node/scriptgenerator.go [hunk 2]
```diff
@@ -47,9 +47,9 @@ const inspectParamVariableName = "ORYX_NODE_INSPECT_PARAM"
 // Checks if the application insights needs to be enabled for the current runtime
 func shouldApplicationInsightsBeConfigured() bool {
 	nodeAppInsightsKeyEnv := os.Getenv("APPINSIGHTS_INSTRUMENTATIONKEY")
-	nodeAppInsightsEnabledEnv := os.Getenv("APPLICATIONINSIGHTSAGENT_EXTENSION_ENABLED")
+	nodeAppInsightsEnabledEnv := os.Getenv("ApplicationInsightsAgent_EXTENSION_VERSION")
 
-	if nodeAppInsightsKeyEnv != "" && strings.ToLower(nodeAppInsightsEnabledEnv) == "true" {
+	if nodeAppInsightsKeyEnv != "" && strings.ToLower(nodeAppInsightsEnabledEnv) == "disabled" {
```
Comment (markwolff) [inline src/startupscriptgenerator/src/node/scriptgenerator.go]: `!= "disabled"`

File: tests/Oryx.RuntimeImage.Tests/Node/NodeRuntimeImageCanRunWhenAppInsightsModuleNotFound.cs (hunk 1 above)
Comment (arroyc) [reply tests/Oryx.RuntimeImage.Tests/Node/NodeRuntimeImageCanRunWhenAppInsightsModuleNotFound.cs]: well im only setting up the extension_version env variable , I didn't set the ai_key .. so I'm expecting my app will run but app insights wont be configured. but I will double check my theory .. Right now I'm building my images locally .. I will test all the tests once the images are built .. in the mean time I created the PR for more visibility towards other people :) .. but thanks for pointing out

File: src/startupscriptgenerator/src/node/scriptgenerator.go (hunk 2 above)
Comment (arroyc) [reply src/startupscriptgenerator/src/node/scriptgenerator.go]: thanks :)

File: src/startupscriptgenerator/src/node/scriptgenerator.go [hunk 3]
```diff
@@ -47,9 +47,9 @@ const inspectParamVariableName = "ORYX_NODE_INSPECT_PARAM"
 // Checks if the application insights needs to be enabled for the current runtime
 func shouldApplicationInsightsBeConfigured() bool {
 	nodeAppInsightsKeyEnv := os.Getenv("APPINSIGHTS_INSTRUMENTATIONKEY")
-	nodeAppInsightsEnabledEnv := os.Getenv("APPLICATIONINSIGHTSAGENT_EXTENSION_ENABLED")
+	nodeAppInsightsEnabledEnv := os.Getenv("ApplicationInsightsAgent_EXTENSION_VERSION")
```
Comment (kichalla) [inline src/startupscriptgenerator/src/node/scriptgenerator.go]: I think it's time to move these strings to `constants.yml`?

File: src/startupscriptgenerator/src/node/scriptgenerator.go [hunk 4]
```diff
@@ -47,9 +47,9 @@ const inspectParamVariableName = "ORYX_NODE_INSPECT_PARAM"
...
 
-	if nodeAppInsightsKeyEnv != "" && strings.ToLower(nodeAppInsightsEnabledEnv) == "true" {
+	if nodeAppInsightsKeyEnv != "" && strings.ToLower(nodeAppInsightsEnabledEnv) !== "disabled" {
```
Comment (kichalla) [inline src/startupscriptgenerator/src/node/scriptgenerator.go]: you meant `!=`

**Pull request merged**